ResponsiveMedia/Drivers/ResponsiveMediaFieldDisplayDriver.cs

```diff
@@ -33,15 +57,61 @@ public override IDisplayResult Edit(ResponsiveMediaField field, BuildFieldEditor
                 model.Field = field;
                 model.Part = context.ContentPart;
                 model.PartFieldDefinition = context.PartFieldDefinition;
-                model.Data = field.Data;
+                model.Data = JsonConvert.SerializeObject(ParseMedia(field.Data));
             });
         }
 
         public override async Task<IDisplayResult> UpdateAsync(ResponsiveMediaField field, IUpdateModel updater, UpdateFieldEditorContext context)
         {
-            await updater.TryUpdateModelAsync(field, Prefix, f => f.Data);
+            if (await updater.TryUpdateModelAsync(field, Prefix, f => f.Data))
+            {
+                var settings = context.PartFieldDefinition.Settings.ToObject<ResponsiveMediaFieldSettings>();
+
+                if (settings.Required && !JsonConvert.DeserializeObject<IList<ResponsiveMediaItem>>(field.Data).Any())
+                {
+                    updater.ModelState.AddModelError(Prefix, S["{0}: Media is required.", context.PartFieldDefinition.DisplayName()]);
+                }
+
+            }
 
             return Edit(field, context);
         }
+
+        public IList<ResponsiveMediaItem> ParseMedia(string data)
```

Comment:
Not saying you have to change this, just want to have a discussion. I quite like having the `Field` entry have the `Model` classes on it directly, in this case having `IList<ResponsiveMediaItem> Data` instead of `string Data` since it's essentially going to store it as JSON anyway it just seems neater to me. In this case though I realise you're going to need to do processing on these when you pull them out anyway to get the correct `Name` and `Url`.

Reply:
Interesting! That may fix an issue I found this morning (https://github.com/EtchUK/Etch.OrchardCore.Fields/pull/11/commits/50d603a92f9c835609b8177049367bd25e77960c).

Think when I initially developed the `ResponsiveMediaField` model, I assumed it would have to store as JSON instead of being able to store a typed list. 

I'm going to look at doing this now.